### PR TITLE
Added a command prompt with CTRL+K

### PR DESCRIPTION
**Pull request by Lauryy06**
image

Same behavior as in (for example) Visual Studio Code. There is :
- Arrow navigation between commands (cyclic)
- Commands filtering
- Enter or click to run command

I've added a "tooltip" field in the `Command` interface, but I don't know the best way to display them, and maybe it's possible to animate the appearance/disappearance of the command prompt.

### CODE REVIEW

File: src/Assets/shadows.sass [hunk 1]
```diff
@@ -0,0 +1,2 @@
+@mixin default-shadow()
+    box-shadow: rgba(0, 0, 0, 0.1) 0 4px 12px
```
Comment (Syycorax) [inline src/Assets/shadows.sass:2]:
```suggestion
    box-shadow: rgba(0, 0, 0, 0.1) 0 4px 12px
    
```
Add newline at the end of file

File: src/Components/CommandPrompt/CommandPrompt.module.sass [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+            .commandName
+                display: inline-block
+
+            .matchedCommandFirstPart
+                margin-left: 12px
+
+            .matchedCommandPart
+                color: colors.$accent
+                font-weight: bold
```
Comment (Syycorax) [inline src/Components/CommandPrompt/CommandPrompt.module.sass:73]:
```suggestion
                font-weight: bold
                
```
Add newline at the end of file

File: src/Models/Command.tsx [hunk 3]
```diff
@@ -0,0 +1,10 @@
+export type CommandCallback<> = () => void
+
+export type Command = {
+    icon?: string
+    id: number
+    name: string
+    tooltip: string
+    // TODO: Add a possible shortcut to commands
+    callback: CommandCallback
+};
```
Comment (Syycorax) [inline src/Models/Command.tsx:10]:
```suggestion
};

```
Add newline at the end of file

**Review by Syycorax — CHANGES_REQUESTED** [review]

File: src/Components/CommandPrompt/CommandPrompt.module.sass (hunk 2 above)
Comment (quentinguidee) [inline src/Components/CommandPrompt/CommandPrompt.module.sass:73]:
```suggestion
                font-weight: $weight-bold
```

File: src/Components/CommandPrompt/CommandPrompt.module.sass [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        display: block
+
+    position: fixed
+    left: 0
+    right: 0
+    top: 0
+    bottom: 0
+    width: 100%
+    height: 100%
+    z-index: 1000
```
Comment (quentinguidee) [inline src/Components/CommandPrompt/CommandPrompt.module.sass:21]:
```suggestion
    z-index: 20
```

File: src/Components/CommandPrompt/CommandPrompt.tsx [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+import styles from "./CommandPrompt.module.sass";
+import Layout from "Components/Layout/Layout";
+import Symbol from "Components/Icon/Symbol";
+import { COMMAND_LIST } from "../../commands";
+import { Command } from "Models/Command";
+
+type Props = React.HTMLProps<HTMLDivElement>;
+
+type State = {
+    shown: boolean
```
Comment (quentinguidee) [inline src/Components/CommandPrompt/CommandPrompt.tsx:13]:
```suggestion
    shown: boolean;
```

**Review by quentinguidee — CHANGES_REQUESTED** [review]

**Comment by Lauryy06** [comment]
See #13

**Pull request closed without merging**